pkg/observable/notifiable/observable.go

```diff
@@ -0,0 +1,93 @@
+package notifiable
+
+import (
+	"context"
+	"sync"
+
+	"pocket/pkg/observable"
+)
+
+// Observable implements the observable.Observable interface and can be notified
+// via its corresponding notifier channel.
+type Observable[V any] struct {
+	mu          sync.RWMutex
+	ch          <-chan V // private channel that is used to emit values to subscribers
+	subscribers []chan V // subscribers is a list of channels that are subscribed to the observable
+	closed      bool
```

Comment:
> What are your thoughts on implementing a function like this:
> 
> ```go
> func (o *Observable[v]) isClosed() bool {
> 	_, ok := <-o.ch
> 	return !ok
> }
> ```

I think as you've written it, it would deadlock if the channel never receives after `#isClosed()` is called.

We considered something similar to this which accounted for that case (i.e. the channel is still open and blocking); however, at the time we somehow missed the fact that `ok` would be false when the channel is closed 😅. Tl;dr, I think it should look like this:

```go
func (o *Observable[v]) isClosed() bool {
  select {
    case _, ok := <-o.ch:
      return !ok
    default:
      return false
}
```

Reply:
Cool. So do you think we should go with the `isClosed` function approach?

Reply:
I tried really hard but was not able to make this work. The issue I was unable to resolve was the fact that `isClosed()` would consume a value from the channel in the case where it's not closed. After some refactoring though I think I've minimized the amount of state and references  thereto.

Reply:
It seems that `select` as it is a non blocking receive would do the trick. If the channel is closed it will always return. Which makes me think that this would suffice
```go
func (o *Observable[v]) isClosed() bool {
    select {
    case <-o.ch:
        return true
    default:
    }
    return false
}
```

Reply:
Agreed. Let's drop this and leave as is.